yamux/src/connection.rs

```diff
@@ -504,23 +502,13 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Active<T> {
                 // The remote may be out of credit though and blocked on
                 // writing more data. We may need to reset the stream.
                 State::SendClosed => {
-                    if self.config.window_update_mode == WindowUpdateMode::OnRead
-                        && shared.window == 0
-                    {
-                        // The remote may be waiting for a window update
-                        // which we will never send, so reset the stream now.
-                        let mut header = Header::data(stream_id, 0);
-                        header.rst();
-                        Some(Frame::new(header))
-                    } else {
-                        // The remote has either still credit or will be given more
-                        // (due to an enqueued window update or because the update
-                        // mode is `OnReceive`) or we already have inbound frames in
-                        // the socket buffer which will be processed later. In any
-                        // case we will reply with an RST in `Connection::on_data`
-                        // because the stream will no longer be known.
-                        None
-                    }
+                    // The remote has either still credit or will be given more
+                    // due to an enqueued window update or we already have
+                    // inbound frames in the socket buffer which will be
+                    // processed later. In any case we will reply with an RST in
+                    // `Connection::on_data` because the stream will no longer
+                    // be known.
+                    None
```

Comment:
@jxs @mxinden, could you please explain why this is the new correct behavior when the window update mode is always OnRead? It seems to be different from the previous behavior.